kyuubi-server/src/main/scala/org/apache/kyuubi/util/KubernetesUtils.scala

```diff
@@ -137,6 +137,8 @@ object KubernetesUtils extends Logging {
     lazy val resolvedResourceName = s"kyuubi-${getResourceNamePrefix(appName, engineRefId)}-driver"
     if (forciblyRewrite || resolvedResourceName.length > DRIVER_POD_NAME_MAX_LENGTH) {
       s"kyuubi-$engineRefId-driver"
+    } else if (appName.contains(engineRefId)){
+      getResourceNamePrefix(appName, "driver")
     } else {
       resolvedResourceName
     }
```

Comment:
```suggestion
    val resourceNamePrefix = if (appName.contains(engineRefId)) {
      getResourceNamePrefix(appName, "")
    } else {
      getResourceNamePrefix(appName, engineRefId)
    }
    val resolvedResourceName = if (resourceNamePrefix.startsWith("kyuubi")) {
      s"$resourceNamePrefix-driver"
    } else {
      s"kyuubi-$resourceNamePrefix-driver"
    }
    if (resolvedResourceName.length <= DRIVER_POD_NAME_MAX_LENGTH) {
      resolvedResourceName
    } else {
      s"kyuubi-$engineRefId-driver"
    }
```

Reply:
> 
Yes,you are right.And I tested with your code and it passed, but about `resourceNamePrefix. startsWith ("kyuubi")`, if  `spark. app. name` is set to the name starting with kyuubi, it will have the effect of the arrow in the photo. So should it be changed to `resourceNamePrefix. startsWith ("kyuubi-")`?
![企业微信截图_17004520968190](https://github.com/apache/kyuubi/assets/45907917/9b7de5c8-29d9-4764-aee1-919d1dad4ec3)